File: .tmp_py4013/batch_17_dump/3937_minimum_operations_to_make_array_modulo_alternating_i.js
// LeetCode 3937 - Minimum Operations To Make Array Modulo Alternating I
// https://leetcode.com/problems/minimum-operations-to-make-array-modulo-alternating-i/
var minOperations = function(nums, k) {
        for (let i = 0; i < nums.length; i++) nums[i] %= k;
        let ans = 2147483647;
        for (let x = 0; x < k; x++) {
            for (let y = 0; y < k; y++) {
                if (x == y) continue;
                let cnt = 0;
                for (let i = 0; i < nums.length; i++) {
                    let target = (i & 1) != 0 ? y : x;
                    let diff = Math.abs(target - nums[i]);
                    cnt += Math.min(diff, k - diff);
                }
                ans = Math.min(ans, cnt);
            }
        }
        return ans;
    
};

File: .tmp_rs4013/rust_check22/3937_minimum_operations_to_make_array_modulo_alternating_i.rs
struct Solution;
// LeetCode 3937 - Minimum Operations To Make Array Modulo Alternating I
// https://leetcode.com/problems/minimum-operations-to-make-array-modulo-alternating-i/

impl Solution {
    pub fn min_operations(mut nums: Vec<i32>, k: i32) -> i32 {
        for v in nums.iter_mut() {
            *v %= k;
        }
        let mut ans = i32::MAX;
        for x in 0..k {
            for y in 0..k {
                if x == y {
                    continue;
                }
                let mut cnt = 0;
                for (i, &num) in nums.iter().enumerate() {
                    let target = if i & 1 == 1 { y } else { x };
                    let diff = (target - num).abs();
                    cnt += diff.min(k - diff);
                }
                ans = ans.min(cnt);
            }
        }
        ans
    }
}

fn main() {}
